grails-app/services/org/pih/warehouse/product/ProductSupplierPreferenceService.groovy

```diff
@@ -0,0 +1,22 @@
+package org.pih.warehouse.product
+
+import grails.gorm.transactions.Transactional
+
+@Transactional
+class ProductSupplierPreferenceService {
+
+    ProductSupplierPreference save(ProductSupplierPreference productSupplierPreference) {
+        // Add the product supplier preference instance to the productSupplier's preferences collection (OneToMany bidirectional)
+        productSupplierPreference?.productSupplier?.addToProductSupplierPreferences(productSupplierPreference)
+        return productSupplierPreference.save(flush: true, failOnError: true)
```

Comment:
`flush` is required here. Since we have validation for uniqueness of `destinationParty` per `productSupplier`, if we create preferences in batch and let's say you send three preferences with the same `destinationParty`, all of them would be saved correctly, because we didn't synchronize the persistence context.
The `flush` is required here to synchronize the DB constraints - in this case - uniqueness of the `destinationParty` per `productSupplier`.